src/test/test_mempool.cc

```diff
@@ -365,6 +365,35 @@ TEST(mempool, bufferlist_reassign)
   ASSERT_EQ(bytes_before, mempool::osd::allocated_bytes());
 }
 
+TEST(mempool, check_shard_select)
+{
+  const size_t samples = 100;
+  std::atomic_int shards[mempool::num_shards] = {0};
+  std::vector<std::thread> workers;
+  for (size_t i = 0; i < samples; i++) {
+    workers.push_back(
+      std::thread([&](){
+          size_t i = mempool::pool_t::pick_a_shard_int();
+          shards[i]++;
+        }));
+  }
+  for (auto& t:workers) {
+    t.join();
+  }
+  workers.clear();
```

Comment:
/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/test_mempool.cc:383:11: error: request for member 'clear' in 'workers', which is of non-class type 'int'
  383 |   workers.clear();
      |           ^~~~~
```
[ 78%] Building CXX object src/librbd/CMakeFiles/rbd_internal.dir/operation/TrimRequest.cc.o
src/test/CMakeFiles/unittest_mempool.dir/build.make:62: recipe for target 'src/test/CMakeFiles/unittest_mempool.dir/test_mempool.cc.o' failed
make[3]: *** [src/test/CMakeFiles/unittest_mempool.dir/test_mempool.cc.o] Error 1
```

Reply:
@trociny Any idea why the nautilus version thinks `workers` is an `int`, here?

Reply:
You should look at the first error reported. This error is a consequence of the previous compiler error, due to which `workers` was not properly defined:
```
/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/test_mempool.cc: In member function 'virtual void mempool_check_shard_select_Test::TestBody()':
/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/test_mempool.cc:372:20: error: 'thread' is not a member of 'std'
  372 |   std::vector<std::thread> workers;
      |                    ^~~~~~
```
So you have to add `#include <thread>`, which I believe exists in the master (or somehow implicitly included via some other headers).

Reply:
@trociny Added it, but that didn't make the error go away :-(

Reply:
> @trociny Added it, but that didn't make the error go away :-(

I eat my words. It **did** make the error go away. But it needed to be the the very last `#include` in the file.